### PR TITLE
executor: let plan replayer support view

### LINKED ISSUE #37163: Plan Replayer couldn't handle View
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```sql
create view revenue0 (supplier_no, total_revenue) as
	select
		l_suppkey,
		sum(l_extendedprice * (1 - l_discount))
	from
		lineitem
	where
		l_shipdate >= '1997-07-01'
		and l_shipdate < date_add('1997-07-01', interval '3' month)
	group by
		l_suppkey;

Plan Replayer Dump Explain select
	s_suppkey,
	s_name,
	s_address,
	s_phone,
	total_revenue
from
	supplier,
	revenue0
where
	s_suppkey = supplier_no
	and total_revenue = (
		select
			max(total_revenue)
		from
			revenue0
	)
order by
	s_suppkey;

drop view revenue0;
```

### 2. What did you expect to see? (Required)

PlanReplayer dumps success

### 3. What did you see instead (Required)

failed

### 4. What is your TiDB version? (Required)

master

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: yisaer <disxiaofei@163.com>

### What problem does this PR solve?

Issue Number: close #37163 

Problem Summary:

### What is changed and how it works?

let plan replayer support view

### Check List

Tests 

- [x] Unit test
- [x] Manual test (add detailed scripts or steps below)

Let plan replayer dump query including view, and load it into other tidb clusters

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/plan_replayer.go [hunk 1]
```diff
@@ -75,10 +81,18 @@ func (tne *tableNameExtractor) Enter(in ast.Node) (ast.Node, bool) {
...
 func (tne *tableNameExtractor) Leave(in ast.Node) (ast.Node, bool) {
+	if tne.err != nil {
+		return in, true
+	}
 	if t, ok := in.(*ast.TableName); ok {
-		tp := tableNamePair{DBName: t.Schema.L, TableName: t.Name.L}
+		isView, err := tne.handleIsView(t)
+		if err != nil {
+			tne.err = err
+			return in, isView
```
Comment (xuyifangreeneyes) [inline executor/plan_replayer.go:91]: Why is the second return values `isView`？

**Review by xuyifangreeneyes — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**